### PR TITLE
Add Bootmagic Lite to QMK

### PR DESCRIPTION
**Pull request by drashna**
This allows for boards to have the "hold while plugging in" option to jump into the bootloader without the headache that full on bootmagic can bring with it.  

Documentation is included on how to set the key in question (row and column), as well as the ability to replace the function to customize it based on the board.

### CODE REVIEW

**Comment by yiancar** [comment]
Could we add a second key and && them together?
I think just one single key to be held down is too little safety.

File: docs/feature_bootmagic.md [hunk 1]
```diff
@@ -99,3 +99,45 @@ If you would like to change the hotkey assignments for Bootmagic, `#define` thes
 |`BOOTMAGIC_KEY_DEFAULT_LAYER_5`         |`KC_5`       |Make layer 5 the default layer                     |
 |`BOOTMAGIC_KEY_DEFAULT_LAYER_6`         |`KC_6`       |Make layer 6 the default layer                     |
 |`BOOTMAGIC_KEY_DEFAULT_LAYER_7`         |`KC_7`       |Make layer 7 the default layer                     |
+
+# Bootmagic Lite
+
+In addition to the full blown bootmagic feature, is the Bootmagic Lite feature that only handles jumping into the bootloader.  This is great for boards that don't have a physical reset button but you need a way to jump into the bootloader, and don't want to deal with the headache that Bootmagic can cause.
```
Comment (mechmerlin) [inline docs/feature_bootmagic.md]:
```suggestion
Please capitalize Bootmagic in the first sentence
```

File: docs/feature_bootmagic.md [hunk 2]
```diff
@@ -99,3 +99,45 @@ If you would like to change the hotkey assignments for Bootmagic, `#define` thes
...
+
+To enable this version of bootmagic, you need to enable it in your `rules.mk` with:
```
Comment (mechmerlin) [inline docs/feature_bootmagic.md]:
```suggestion
Capitalize bootmagic
```

File: docs/feature_bootmagic.md [hunk 3]
```diff
@@ -99,3 +99,45 @@ If you would like to change the hotkey assignments for Bootmagic, `#define` thes
...
+    matrix_scan();
+
+    if (matrix_get_row(BOOTMAGIC_LITE_ROW) & (1 << BOOTMAGIC_LITE_COLUMN)) {
+      // Jump to bootloader.
+      bootloader_jump();
+    }
+}
+```
+
+You can additional feature here like resetting the eeprom, or the like.  Keep in mind that `bootmagic_lite` is called before a majority of features are initialized in the firmware.
```
Comment (mechmerlin) [inline docs/feature_bootmagic.md]:
```suggestion
"You can add additional features here...."
What do you mean by "or the like". 
```

**Comment by Wilba6582** [comment]
@yiancar there are hundreds of M6-A's out there with only a single key bootmagic_lite, you worry too much, friendo ;-)

But I agree with you on the EEPROM reset, that's the #1 thing that solves Zeal60 customer issues, because of some random bit of QMK magic being stuck on...

File: docs/feature_bootmagic.md (hunk 1 above)
Comment (drashna) [reply docs/feature_bootmagic.md]: @mechmerlin You want to post the corrected text in the suggestion, actually.  But yes. 

```suggestion
In addition to the full blown Bootmagic feature, is the Bootmagic Lite feature that only handles jumping into the bootloader.  This is great for boards that don't have a physical reset button but you need a way to jump into the bootloader, and don't want to deal with the headache that Bootmagic can cause.
```

File: docs/feature_bootmagic.md (hunk 2 above)
Comment (drashna) [reply docs/feature_bootmagic.md]:
```suggestion
To enable this version of Bootmagic, you need to enable it in your `rules.mk` with:
```

File: docs/feature_bootmagic.md (hunk 3 above)
Comment (drashna) [reply docs/feature_bootmagic.md]:
```suggestion
You can additional feature here. For instance, resetting the eeprom or requiring additional keys to be pressed to trigger bootmagic.  Keep in mind that `bootmagic_lite` is called before a majority of features are initialized in the firmware.
```

**Comment by drashna** [comment]
@yiancar I could, and that's what I had originally. But Wilba has a point about the single key stuff.

Additionally, if you noticed, I defined the function weakly.  So if you want 2 keys required on your board, you can add `bootmagic_lite` to your keyboard's c file, and it will use that function instead.  This allows for customization of the feature.... up to the point that you could emulate full bootmagic. 

@mechmerlin I thought about that, but the problem with that, is then I need to check both option's status and error out if both are enabled.  This is a simpler way to ensure that only one of them is enabled, and with minimal change to keymaps. 

I'm not sure which approach is best, though.  I personally like this one better, though. 
@jackhumbert @skullydazed @fredizzimo, any input on how to handle enabling this feature?

**Comment by yiancar** [comment]
Yep I saw that so no complains for the single key! However the eeprom is a must I think:)

**Comment by drashna** [comment]
@yiancar I'm not sold on the eeprom reset. 

In fact, I found it more of a pain in the ass, for me. 

However, I can re-add it, if there is a good reason for it.

**Comment by jackhumbert** [comment]
Great! Thanks :)

**Pull request merged**